packages/editor-ui/src/components/SqlEditor/SqlEditor.vue

```diff
@@ -189,18 +190,29 @@ export default defineComponent({
 		onBlur() {
 			this.isFocused = false;
 		},
-		highlightLine(line: number | 'final') {
+		line(lineNumber: number): Line | null {
+			try {
+				return this.editor?.state.doc.line(lineNumber) ?? null;
+			} catch {
+				return null;
+			}
+		},
+		highlightLine(lineNumber: number | 'final') {
```

Comment:
Removing the duplication introduced by the SQL editor is out of scope for this fix.